internal/hardware/versioned-requirements.go

```diff
@@ -0,0 +1,42 @@
+package hardware
+
+import "encoding/json"
+
+type Requirements struct {
```

Comment:
Maybe declare this struct in `swagger.yaml`? This will give us this code for free, and might allow this struct to be reused / created by outside components

Reply:
WDYT about reusing https://github.com/openshift/assisted-service/blob/e231612a06de5d6bb49d586729676f8474c7ba5d/swagger.yaml#L3237? 
There is a difference in units - cluster-host-requirements-details use MiB for memory.

Reply:
Personally I don't see a reason not to use that struct.

If you'll move to this new struct, what about `installation_disk_speed_threshold_ms`? -

It appears in that struct and I believe it might also change between OpenShift versions, so maybe you should handle that as well.

The current threshold we picked for it is taken from the OpenShift docs, IIRC.

Maybe discuss this with @ori-amizur

Reply:
Yes, that would require it to be handled as well. 
From what I can see in the code, `installation_disk_speed_threshold_ms` is not used for any validation so far and it isn't overriden with env variables (with default of 0).

Reply:
Yes, right now it's a hard coded threshold in the agent, but @ori-amizur is working on moving that validation to the service

Reply:
Changed the config to use `models.ClusterHostRequirementsDetails`

Reply:
Sorry, my intention was that also the entire `VersionedRequirements` struct should be in `swagger.yaml` - this way the serializing / deserializing code and some tests come for free

Reply:
Of course; ma bad. :facepalm: